### DISCUSSION TITLE
Minimize docker build - what files to exclude

### DISCUSSION THREAD

**Discussion by JardarTemalogic** [prisma/prisma#16624] [Q&A]
Hi,

by default when building the application for production prisma occupies several hundred MB. From what I understand most of the binary files can be removed before building the docker container. I guess that this will be solved properly in the future, but it would be great if somebody could provide a list of files that can be explicitly removed.
My production build is only going to query the database and not migrate it.

**Comment by nurul3101** [comment]
Hello @JardarTemalogic 👋 

A Similar GitHub discussion discusses solutions to minimize the docker size: [#11572](https://github.com/prisma/prisma/discussions/11572). 
You can delete anything in the `.cache` directory if you want to save some space.
Also, here's a GitHub Issue: [#11577](https://github.com/prisma/prisma/issues/11577) which tracks reducing the docker build size.

**Comment by JardarTemalogic** [comment]
Thanks nurul, I know about these tickets however it does not state exactly what can be removed. I know I can remove .cached, but did some more tests and was able to remove the following and it was still working.

RUN rm -rf node_modules/.cache/
RUN rm -rf node_modules/@prisma/engines/*.exe
RUN rm -rf node_modules/@prisma/engines/introspection*
RUN rm -rf node_modules/@prisma/engines/migration*
RUN rm -rf node_modules/@prisma/engines/prisma-fmt*
RUN rm -rf node_modules/.prisma/client/query_engine-windows.dll.node

Thus it would be good if a "offical" Dockerfile suggestion regarding what components that could be removed to minimize the build.

**Reply by nurul3101** [reply]
That's good to know, can you add this comment in this GitHub Issue: https://github.com/prisma/prisma/issues/11577?
Our engineering team monitors the issue, and they will be able to provide suggestions for this.

**Reply by JardarTemalogic** [reply]
Sure, I added the content of my Dockerfile in #11577

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
